crates/sparrow-sources/src/in_memory.rs

```diff
@@ -174,9 +177,7 @@ impl InMemoryBatches {
     ///
     /// The first batch will be the in-memory merged batch, and batches will be
     /// added as they arrive.
-    pub fn subscribe(
-        &self,
-    ) -> impl Stream<Item = error_stack::Result<RecordBatch, SourceError>> + 'static {
+    pub fn subscribe(&self) -> BoxStream<'static, error_stack::Result<RecordBatch, SourceError>> {
```

Comment:
This is probably fine, although I suspect we're boxing it possibly again later (which we may want to check for and remove, if we are). Is there a reason it needed to be boxed?

Reply:
...I guess not. I would have sworn that I was unable to call `next()` on it because it wasn't `Unpin`, but maybe I hallucinated that.